kai/analyzer.py

```diff
@@ -109,18 +113,27 @@ def __init__(
     @tracer.start_as_current_span("run_analysis")
     def run_analyzer_lsp(
         self,
-        label_selector: str,
-        included_paths: list[Path],
-        incident_selector: str,
+        label_selector: Optional[str] = None,
+        included_paths: Optional[list[Path]] = None,
+        incident_selector: Optional[str] = None,
         scoped_paths: Optional[list[Path]] = None,
         reset_cache: Optional[bool] = None,
     ) -> JsonRpcResponse | JsonRpcError | None:
+
+        if label_selector is not None and self.labels is None:
+            self.labels = label_selector
+
+        if label_selector is None and self.labels is not None:
+            label_selector = self.labels
+
+        if incident_selector is not None:
+            self.incident_selector = incident_selector
+
```

Comment:
Nit: Do we want to statefully modify the class here or should we use the class as the default and have the params as modifications?

Reply:
The idea is that if they are not set on startup, but set after, and then you run analysis, we capture them and save them for later (specifically in the case that you then run code plan).